### PR TITLE
Attempts to Improve CPS Behavior

### LINKED ISSUE #38: Enhance CPS Behavior
Right now CPS works but its pretty rudimentary for instance if CPS is running and a SQRL login action is started the browser hangs (as it should) waiting for a CPS Response.
However if the SQRL client closes or crashes the browser just goes to 404

We should handle that gracefully on exit any pending CPS requests need to be redirected to the cancel URL and things of that sort

### LINKED ISSUE #126: [Linux] Idle timeout seems to be broken
While testing on Linux, I noticed log messages regarding idle timeout being hit, even though the timeout in the loaded identity was set to 15 minutes. It seems that the idle timeout is consistently hit after about 5 seconds. Something seems broken here.

### PR DESCRIPTION
**Pull request by josegomez**
This enhances CPS Behavior to re-direct to a generic page if the cancel token is missing or something fails (program exists out of cycle)

This won't/can't catch big major SIGTERM style crashes but I believe it is the best we can do since the OS is King
CPSEnhancemets

@alexhauser The new german strings need some help 😋😆

### CODE REVIEW

**Comment by alexhauser** [comment]
Will review asap @josegomez, am afk atm.

File: SQRLDotNetClientUI/Assets/Localization/localization.json [hunk 1]
```diff
@@ -468,6 +474,9 @@
     { "OrLabel": "... oder ..." },
     { "AppSettingsMenuHeader": "Einstellungen" },
     { "AppSettingsTitle": "Allgemeine Einstellungen" },
-    { "StartMinimizedLabel": "Programmfenster beim Start minimieren" }
+    { "StartMinimizedLabel": "Programmfenster beim Start minimieren" },
+    { "CPSAbortHeader": "Authentication Aborted" },
```
Comment (alexhauser) [inline SQRLDotNetClientUI/Assets/Localization/localization.json]: "Authentifizierung abgebrochen"

File: SQRLDotNetClientUI/Assets/Localization/localization.json [hunk 2]
```diff
@@ -468,6 +474,9 @@
...
+    { "CPSAbortMessage": "SQRL's CPS authentication has been aborted. You will be automatically sent back to the previous page in a few seconds. If this does not work, please press your browser's BACK button or click the link below." },
```
Comment (alexhauser) [inline SQRLDotNetClientUI/Assets/Localization/localization.json]: "SQRLs CPS-Authentifizierung wurde abgebrochen. Sie werden in ein paar Sekunden automatisch zur vorherigen Seite zurückgeleitet. Falls dies nicht funktioniert, drücken Sie bitte in Ihrem Browser auf die Schaltfläche \"Zurück\" oder klicken Sie auf den folgenden Link."

File: SQRLDotNetClientUI/Assets/Localization/localization.json [hunk 3]
```diff
@@ -468,6 +474,9 @@
...
+    { "CPSAbortLinkText": "Go Back Now" }
```
Comment (alexhauser) [inline SQRLDotNetClientUI/Assets/Localization/localization.json]: "Jetzt zurückgehen"

File: SQRLDotNetClientUI/Program.cs [hunk 4]
```diff
@@ -79,6 +86,27 @@ public static void Main(string[] args)
...
+                        else
+                            SQRLCPSServer.HandlePendingCPS();
+                    }
+                    catch(Exception ex)
+                    {
+                        Log.Error("Failed to Cancel CPS Gracefully", ex);
+                    }
+
+
+
```
Comment (alexhauser) [inline SQRLDotNetClientUI/Program.cs]: This code almost exactly duplicates what's already in `CurrentDomain_ProcessExit()`. Can we make one general function out of it and call it in both places?

File: SQRLUtilsLib/SQRL.cs [hunk 5]
```diff
@@ -42,7 +42,10 @@ public class SQRL
         private const int ENCODING_BASE = 56;
         private const int CLIENT_VERSION = 1;
 
-        public CPSServer cps=null;
+        
+
+
+        public SQRLCPSServer cps=null;
```
Comment (alexhauser) [inline SQRLUtilsLib/SQRL.cs:48]: I like it that you renamed this to `SQRLCPSServer` to be in line with all the other SQRLxxx classes in the library!

File: SQRLUtilsLib/SQRLCPSServer.cs [hunk 6]
```diff
@@ -57,13 +58,44 @@ public class CPSServer
         /// </summary>
         public bool Running = false;
 
+        /// <summary>
+        /// Holds the HTML Content of the CPS Abort page pulled from an internal Resource File
+        /// </summary>
+        private string AbortHTML = "";
```
Comment (alexhauser) [inline SQRLUtilsLib/SQRLCPSServer.cs]: If possible, I would like to keep our coding style consistent throughout the project.
In the client, I have meanwhile harmonized almost all the code to follow a single style, maybe we can do so in the library too:

Private fields should start with underscore and a small letter: `private string _abortHTML = ""; `

File: SQRLUtilsLib/SQRLCPSServer.cs [hunk 7]
```diff
@@ -205,5 +257,56 @@ private byte[] GenerateGif()
             const string clearGif1X1 = "R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==";
             return Convert.FromBase64String(clearGif1X1);
         }
+
+
+        /// <summary>
+        /// Handles CPS response for all requests
+        /// </summary>
+        /// <param name="succesUrl">Success URL passed in if the response is a successful Auth Ident</param>
```
Comment (alexhauser) [inline SQRLUtilsLib/SQRLCPSServer.cs:268]: Maybe document the other parameters, too?

**Review by alexhauser — APPROVED** [review]
Nice job @josegomez, looking forward to try this out!
And sorry for the nitpicking!

File: SQRLUtilsLib/SQRLCPSServer.cs (hunk 7 above)
Comment (josegomez) [reply SQRLUtilsLib/SQRLCPSServer.cs:268]: ah yes I added those later LoL on it!

**Comment by josegomez** [comment]
@alexhauser I made the enhancements, will merge when checks are done, includes fix for Linux Timeout

File: SQRLUtilsLib/SQRL.cs (hunk 5 above)
Comment (josegomez) [reply SQRLUtilsLib/SQRL.cs:48]: Thanks! I pay attention (sometimes hehe)

**Comment by alexhauser** [comment]
Great! Sure, go ahead, @josegomez!

**Pull request merged**